### DISCUSSION TITLE
useContractRead and readContract spamming RPC

### DISCUSSION THREAD

**Discussion by LPSCRYPT** [wevm/wagmi#1727] [General]
I'm trying to implement either `useContractRead` or `readContract`, however I run into an issue that every time I call these functions, they spam the RPC, causing a rate limiting error. I can see in the console that I'm getting multiple `429` errors from the RPC spamming. I'm certain that the functions are only being called once: at least in case of `readContact`, as I can wrap it in a button push and run other checks which verify that it is only called once, yet still leading to RPC spamming. Here are examples of the call implementation: 

`	const { data: callData } = useContractRead({
		addressOrName: "0xdac17f958d2ee523a2206206994597c13d831ec7",
		contractInterface: erc20abi,
		functionName: "getOwner",
		chainId: 1,
	});`

and

`	const { data } = await readContract({
		addressOrName: espgoerli.memberpointsregistry,
		contractInterface: DxDMemberPointsRegistry,
		functionName: "getUserPoints",
		chainId: chainId.goerli,
		args: [1, address]
	});`

Upon deploying the app to localhost, the first call returns correctly without spamming `post` calls to the rpc. However, subsequent page refreshes / triggers cause the rpc to be spammed each time. Killing and redeploying the app restarts this cycle. My provider is configured as follows:

`const wagmiClient = createClient({
	autoConnect: true,
	provider: getDefaultProvider("goerli", { pollingInterval: 15000 })
});`

It seems that any manual call to the provider results in the same behavior (noting the exception that useContractWrite calls do not cause an rpc spam error). For example, a `const net = await provider.getNetwork();` will cause the rpc to be spammed.

Any ideas?

**Comment by SaadNiaziDev** [comment]
```js
enabled: false,
```

place this in readContract and then use refetch function on component/page mount.

```js
const {refetch,data} = useContractRead();

useEffect(()=>{
refetch();
},[]);
